src/helpers/range.ts

```diff
@@ -194,3 +195,43 @@ export function copyRangeWithNewSheetId(sheetIdFrom: UID, sheetIdTo: UID, range:
 export function createRange(getters: CoreGetters, sheetId: UID, range?: string): Range | undefined {
   return range ? getters.getRangeFromSheetXC(sheetId, range) : undefined;
 }
+
+/**
+ * Spread multiple colrows zone to one row/col zone and add a many new input range as needed.
+ * For example, A1:B4 will become [A1:A4, B1:B4]
+ */
+export function spreadRange(ranges: string[]): string[] {
```

Comment:
I think we introduced some behaviour recently that made the assumption that datasets would only be served as columns. This function still raises the question of unbounded zones, how should we handle them? I need to look further into it, we might have gone wrong with a recent spec

In any case, the docstring is incomplete, we don't really know how the spread will take place (at first glance ,I would have sworn it was spitting by column but turns out I wa wrong)